cli/src/lockfiles/javascript.rs

```diff
@@ -102,30 +102,53 @@ impl Parseable for YarnLock {
                 .filter(|s| !s.is_empty())
                 .ok_or_else(|| anyhow!("Failed to parse resolution field in yarn lock file"))?;
 
-            // Ignore workspace-local dependencies like project itself ("project@workspace:.").
-            if resolution[1..].contains("@workspace:") {
-                continue;
+            let (name, mut resolver) = match resolution[1..].split_once('@') {
+                Some((name, resolver)) => (&resolution[..name.len() + 1], resolver.to_owned()),
+                None => return Err(anyhow!("Failed to parse yarn depenency resolution")),
+            };
+
+            // Extract original resolver from patch.
+            if let Some((_, patch)) = resolver.split_once("patch:") {
+                // Exctract resolver from `@scope/package@RESOLVER#patch`.
+                let patch = patch[1..].split_once('@');
+                let subresolver = patch.and_then(|(_, resolver)| resolver.split_once('#'));
+                resolver = match subresolver {
+                    Some((resolver, _)) => resolver.to_owned(),
+                    None => return Err(anyhow!("Failed to parse yarn patch dependency")),
```

Comment:
Some context here might be good.

```suggestion
                    None => return Err(anyhow!("Failed to parse yarn patch dependency for package '{}'", name)),
```